operator/v1/types_ingress.go

```diff
@@ -165,6 +165,26 @@ const (
 	PrivateStrategyType EndpointPublishingStrategyType = "Private"
 )
 
+// LoadBalancerType is a type of load balancer.
+type LoadBalancerType string
+
+var (
+	// InternalLoadBalancer is a load balancer that is exposed only on the
+	// cluster's private network.
+	InternalLoadBalancer LoadBalancerType = "Internal"
+
+	// ExternalLoadBalancer is a load balancer that is exposed on the
+	// cluster's public network (which is typically on the Internet).
+	ExternalLoadBalancer LoadBalancerType = "External"
+)
+
+// LoadBalancerStrategy holds parameters for a load balancer.
+type LoadBalancerStrategy struct {
+	// type indicates the type of load balancer.  Possible values are
+	// "External" and "Internal".  The default is "External".
+	Type LoadBalancerType `json:"type"`
```

Comment:
I'm trying to think of any other possible future types which might be in conflict with this. For example, is there some other "type" of load balancer that must also be internal? If so, would that indicate type is too broad a term? Compared to say, "scope" or "visibility"?

Reply:
We went with a string type instead of a Boolean value because we don't know what possible future types there may be.  GCP uses the annotation `cloud.google.com/load-balancer-type=Internal`, AWS uses `service.beta.kubernetes.io/aws-load-balancer-internal=0.0.0.0/0`, and the other clouds use variations on `somethingsomething-internal-somethingsomething=true`.

Reply:
I was probably unclear. I wasn't suggesting a boolean type for `Type`, I was wondering if `Type` should be `Scope` or `Visibility` or something to that effect.

Reply:
For compatibiliy with https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190325-unions.md, you need structs for each discriminator value:

```golang
Type LoadBalancerType `json:"type"`
External *External `json:"external,omitempty"`
Internal *Internal `json:"internal,omitempty"`
```

I don't think discriminator only (or partial sub-structs) will work.

Reply:
Is Internal/External a _type_ of LoadBalancerStrategy as opposed to (for example) a _property of_ or _a policy applied to_ a LoadBalancerStrategy? Seems to me more property-like (hence Scope or Visibility as alternatives to Type).